kudos-services/src/main/java/org/exoplatform/kudos/listener/KudosActivityListener.java

```diff
@@ -46,11 +54,31 @@ public void updateActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
     }
   }
 
+  @Override
+  public void deleteActivity(ActivityLifeCycleEvent activityLifeCycleEvent) {
+    ExoSocialActivity activity = activityLifeCycleEvent.getSource();
+    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
+    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
+      return;
+    }
+    deleteLinkedKudos(linkedKudosList);
```

Comment:
What if the activity has sub kudos ? It's better to search for all kudos in the activity and cancel them.

```suggestion
    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
    deleteLinkedKudos(linkedKudosList);
    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
      return;
    }
```

or if you don't mean to cancel all kudos comment, when deleting an activity, then just retrieve the list after the condition test:

```suggestion
    if (!StringUtils.equals(activity.getType(), KUDOS_ACTIVITY_COMMENT_TYPE) && linkedKudosList.isEmpty()) {
      return;
    }
    List<Kudos> linkedKudosList = kudosService.getKudosListOfActivity(activity.getId());
    deleteLinkedKudos(linkedKudosList);
```